Move the metrics collection documents (#3533)
SumoLogic/sumologic-documentation: docs/send-data/kubernetes/collecting-kubernetes-metrics.md

> ---

Reviewer: I wonder if we shouldn't just merge this document with the application metrics one and call it `Collecting Metrics`. This one is pretty short and light on content.

id: collecting-kubernetes-metrics
title: Collecting Kubernetes Metrics
sidebar_label: Collecting Kubernetes Metrics
description: Collecting Kubernetes Metrics
---

By default, we collect selected metrics from the following Kubernetes components:

- `Kube API Server` configured with `kube-prometheus-stack.kubeApiServer.serviceMonitor`
- `Kubelet` configured with `kube-prometheus-stack.kubelet.serviceMonitor`
- `Kube Controller Manager` configured with `kube-prometheus-stack.kubeControllerManager.serviceMonitor`
- `CoreDNS` configured with `kube-prometheus-stack.coreDns.serviceMonitor`
- `Kube EtcD` configured with `kube-prometheus-stack.kubeEtcd.serviceMonitor`
- `Kube Scheduler` configured with `kube-prometheus-stack.kubeScheduler.serviceMonitor`
- `Kube State Metrics` configured with `kube-prometheus-stack.kube-state-metrics.prometheus.monitor`
- `Prometheus Node Exporter` configured with `kube-prometheus-stack.prometheus-node-exporter.prometheus.monitor`

> If you want to forward additional metric from one of these services, you need to edit the corresponding Service Monitor definition. Service
> Monitor tells Prometheus which metrics it should take from the service.

Reviewer: ```suggestion
If you want to forward additional metric from one of these services, you need to edit the corresponding Service Monitor definition. Service
Monitor tells the OpenTelemetry Collector which metrics it should take from the service.
```

> ## Example

Reviewer: This example doesn't work anymore, as otel collector doesn't collect cAdvisor metrics based on a ServiceMonitor. We should change this to kube-state-metrics imo.

In addition to all metrics we send by default from CAdvisor you also want to forward `container_blkio_device_usage_total`.

You need to modify `kube-prometheus-stack.kubelet.serviceMonitor.cAdvisorMetricRelabelings` to include `container_blkio_device_usage_total`.

```yaml
kube-prometheus-stack:
  kubelet:
    serviceMonitor:
      cAdvisorMetricRelabelings:
        - action: keep
          regex: (?:container_blkio_device_usage_total|container_cpu_usage_seconds_total|container_memory_working_set_bytes|container_fs_usage_bytes|container_fs_limit_bytes|container_cpu_cfs_throttled_seconds_total|container_network_receive_bytes_total|container_network_transmit_bytes_total)
          sourceLabels: [__name__]
        - action: labelmap
          regex: container_name
          replacement: container
        - action: drop
          sourceLabels: [container]
          regex: POD
        - action: labeldrop
          regex: (id|name)
```

:::note
You can use the method described in
[Troubleshooting Collection document](/docs/send-data/kubernetes/troubleshoot-collection#check-the-metrics-endpoint-for-kubernetes-services) to troubleshoot this process.
:::

## Metrics modifications

> See the relevant section in [Collecting application metrics](/docs/send-data/kubernetes/collecting-application-metrics#metrics-modifications).

Reviewer 2: ```suggestion
See the [Metrics modifications](/docs/send-data/kubernetes/collecting-application-metrics#metrics-modifications) section in [Collecting Application Metrics](/docs/send-data/kubernetes/collecting-application-metrics).
```

## Investigation

> For investigation you may want to look at the
> [Collecting application metrics](/docs/send-data/kubernetes/collecting-application-metrics#metrics-modifications) and
> [Troubleshooting Collection document](/docs/send-data/kubernetes/troubleshoot-collection.md#collecting-metrics)

Reviewer 2: ```suggestion
For investigation you may want to look at [Collecting Application Metrics](/docs/send-data/kubernetes/collecting-application-metrics#metrics-modifications) and [Troubleshooting Collection](/docs/send-data/kubernetes/troubleshoot-collection.md#collecting-metrics).
```
